internal/abi/fastly/types.go

```diff
@@ -127,17 +128,31 @@ func (s FastlyStatus) toError() error {
 	}
 }
 
+func (s FastlyStatus) toErrorDetailed(d sendErrorDetail) error {
+	switch s {
+	case FastlyStatusOK:
+		return nil
+	default:
+		return FastlyError{Status: s, detail: d}
+	}
+}
+
 // FastlyError decorates error-class FastlyStatus values and implements the
 // error interface.
 //
 // Note that TinyGo currently doesn't support errors.As. Callers can use the
 // IsFastlyError helper instead.
 type FastlyError struct {
 	Status FastlyStatus
+	detail sendErrorDetail
```

Comment:
I wonder if we want to replace this with a more generic `error` variable or type that we can use to wrap different Fastly errors as they become available so this struct doesn't end up with a field for each different type of Fastly error returns.

Reply:
Yeah, or conversely we might want to expose the error tags (and I guess the TLS Alert codes) to callers to be able to switch on that info